src/App.tsx

```diff
@@ -1,39 +1,68 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state:State = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  clockId = 0;
```

Comment:
It may be better to give a more descriptive name, for example `clockNameTimerId`